### PR TITLE
Add developer guide

### LINKED ISSUE #52: Add developer guide
The instructions in the [`sdk/README.md`](https://github.com/kubeflow/kfp-tekton/blob/master/sdk/README.md#steps) currently only provide very basic instructions to set up the KFP-Tekton compiler.

The "Developer Guide" will contain more detailed instructions to set up a development environment and explain the structure, concepts and ideas of the compiler code.

### PR DESCRIPTION
**Pull request by ckadner**
Add first version of the KFP-Tekton Developer Guide.

This initial version contains the following topics:

---
  - [Development Prerequisites](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#development-prerequisites)
    - [Installing Tekton](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#installing-tekton)
      - [Tekton Cluster](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#tekton-cluster)
      - [Tekton CLI](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#tekton-cli)
      - [Tekton Dashboard](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#tekton-dashboard)
  - [Origins of the KFP-Tekton Compiler Code](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#origins-of-the-kfp-tekton-compiler-code)
  - [Adding New Code](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#adding-new-code)
    - [Overriding KFP Compiler Methods](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#overriding-kfp-compiler-methods)
    - [Monkey-Patching Static KFP Compiler Methods](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#monkey-patching-static-kfp-compiler-methods)
  - [Coding Style](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#coding-style)
  - [Testing](#testing)
    - [Unit Tests](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#unit-tests)
    - [End-to-End Tests with Tekton](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#end-to-end-tests-with-tekton)
    - [Compiler Test Report](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#compiler-test-report)
  - [License Headers](https://github.com/ckadner/kfp-tekton/blob/add_developer_guide/sdk/python/README.md#license-headers)
---
Resolves #52

### CODE REVIEW

File: scripts/mdtoc.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (Tomcli) [inline scripts/mdtoc.sh:1]: Maybe we should put this script under `/tools` or `/test` to align with Kubeflow pipeline?

File: sdk/python/README.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+<!-- END of ToC generated by running ./scripts/mdtoc.sh sdk/python/README.md -->
+
+
+## Development Prerequisites
+
+1. [`Python`](https://www.python.org/downloads/): version `3.5` or later
+2. [`Kubernetes` Cluster](https://v1-15.docs.kubernetes.io/docs/setup/): version `1.15` ([required by Kubeflow](https://www.kubeflow.org/docs/started/k8s/overview/) and Tekton 0.11)
+3. [`kubectl` CLI](https://kubernetes.io/docs/tasks/tools/install-kubectl/): required to deploy Tekton pipelines to Kubernetes cluster
+4. [`Tekton` Deployment](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0/): version `0.11.0` (or greater to support Tekton API version `v1beta1`), required for end-to-end testing
```
Comment (Tomcli) [inline sdk/python/README.md:37]: maybe we should use v0.11.3 to support the new condition passing?

File: scripts/mdtoc.sh (hunk 1 above)
Comment (ckadner) [reply scripts/mdtoc.sh]: good idea, I moved it to tools

File: sdk/python/README.md (hunk 2 above)
Comment (ckadner) [reply sdk/python/README.md]: yup, I updated it, thanks

**Pull request merged**